tepeszCodes/document-management-system
# Chore: Add logging middleware

#### What does this PR do?
Adds logging middleware to our project. It uses winston.js and morgan.js.

#### Description of Task to be completed?
Add logging middleware

#### How should this be manually tested?
Run `npm run dev` script. `logsDev.log` file should appear with info about the port on which the server started listening. The same info should appear in the stdout console. Send a request to the running server by accessing the port on which Express app is running via your browser. After you do that, info about your request should be added to `logsDev.log` file.

#### Questions:
1. The documentation I found both about winston and morgan is pretty scarce and doesn't explain too much about how those libraries actually work. Also, I didn't find any comprehensive reference doc for either library, which forced me to do a lot of copying, pasting and experimentation. Do you know of any such resources?
2. In the `app.js` if I put ``` 
app.get('/', (req, res) => {
  res.send('This is my first Express server');
});
app.use(morganMiddleware);
```
instead of
```
app.use(morganMiddleware);
app.get('/', (req, res) => {
  res.send('This is my first Express server');
});
```
the `morganMiddleware` is never called. But in the branch [chore/add-error-handling](https://github.com/tepeszCodes/document-management-system/tree/chore/add-error-handling) I call error handling middleware after `app.get(...)` and everything works fine - why is that?
